rex/sam_resource.py

```diff
@@ -701,6 +701,86 @@ def append_var_list(self, var):
 
         self.var_list.append(var)
 
+    def bias_correct(self, bc_df):
+        """Bias correct wind or irradiance data using a table of linear
+        correction factors per resource gid.
+
+        Parameters
+        ----------
+        bc_df : None | pd.DataFrame
+            None if not provided or extracted DataFrame with wind or solar
+            resource bias correction table. This has columns: gid, adder,
+            scalar. If either adder or scalar is not present, scalar defaults
+            to 1 and adder to 0. Only windspeed, GHI, and DNI are corrected.
+            GHI and DNI are corrected with the same correction factors.
+        """
+
+        if 'adder' not in bc_df:
+            logger.info('Bias correction table provided, but "adder" not '
+                        'found, defaulting to 0.')
+            bc_df['adder'] = 0
+
+        if 'scalar' not in bc_df:
+            logger.info('Bias correction table provided, but "scalar" not '
+                        'found, defaulting to 1.')
+            bc_df['scalar'] = 1
+
+        if bc_df.index.name != 'gid':
+            msg = ('Bias correction table must have "gid" column but only '
+                   'found: {}'.format(list(bc_df.columns)))
+            assert 'gid' in bc_df, msg
+            bc_df = bc_df.set_index('gid')
+
+        site_arr = np.array(self.sites)
+        isin = np.isin(self.sites, bc_df.index.values)
+        if not isin.all():
+            missing = site_arr[isin]
+            msg = ('{} sites were missing from the bias correction table, '
+                   'not bias correcting: {}'.format(len(missing), missing))
+            logger.warning(msg)
+            warn(msg)
+
+        scalar = np.expand_dims(bc_df.loc[site_arr[isin], 'scalar'].values, 0)
+        adder = np.expand_dims(bc_df.loc[site_arr[isin], 'adder'].values, 0)
+
+        if 'windspeed' in self._res_arrays:
```

Comment:
This *almost* does not work for NSRD. Are you intentionally relying on the implementation detail of solar data to include an underscore in the variable name (i.e. `wind_speed` instead of `windspeed`)? It's possible this is a standard I am not aware of, but it could lead to problems down the line otherwise. 

One solution that may not be too much extra work to implement is to ask explicitly for a `wind` or a `solar` bc_df (or both!). Then in reV, users could just specify `bias_correct={"wind": "/path/to/wind/bc_df.csv", "solar": "/path/to/solar/bc_df.csv"}`, and the `_parse_bc` function could take care of loading in the DataFrames to pass to this function.

Anyways, I won't insist on anything but just some food for thought

Reply:
Oh wow haha you're totally right, this could have easily went to wind for solar data and skipped the irrad vars. Easier fix is just to check for 'ghi' first which is the more precise condition.